src/components/QueryEditor.test.tsx

```diff
@@ -32,7 +35,7 @@ const dimensions = ['region', 'zone'];
 
 beforeEach(() => {
   ds.getResource = jest.fn().mockResolvedValue(databases);
-  ds.postResource = jest.fn((r: string, body?: any) => {
+  ds.postResource = jest.fn<Promise<any>, [path: string, body?: any]>((r: string, body?: any) => {
```

Comment:
```suggestion
  ds.postResource = jest.fn().mockImplementation((r: string, body?: any) => {
```

Ran into some similar code with my PR https://github.com/grafana/athena-datasource/pull/198/files#diff-f5855f08029d58a7871f69a0ebfe07376b2508725c81159df890fc793eb5b545R67-R79, and just by chance I noticed I didn't need to add types manually. The only difference was using `mockImplementation`.

According to the docs https://jestjs.io/docs/mock-function-api#mockfnmockimplementationfn

> jest.fn(implementation) is a shorthand for jest.fn().mockImplementation(implementation).